pkg/controller/installation/validation.go

```diff
@@ -155,8 +167,14 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 					v6pool.NATOutgoing, strings.Join(operatorv1.NATOutgoingTypesString, ","))
 			}
 
-			if v6pool.NodeSelector == "" {
-				return fmt.Errorf("ipPool.nodeSelector should not be empty")
+			if instance.Spec.CNI.Type == operatorv1.PluginCalico {
+				if v6pool.NodeSelector == "" {
```

Comment:
This validation isn't Calico specific, right? The node-selector must _always_ be non empty?

Reply:
You're right.

Reply:
I'm going to change the IPv4 code also. In both cases I'll move the `NodeSelector == ""` check to after Plugin check block. So in the case of non-Calico we'll check the NodeSelector twice.